csharp/Ice/Cancellation/Client/Program.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) ZeroC, Inc.
+
+// Slice module VisitorCenter in Greeter.ice maps to C# namespace VisitorCenter.
+using VisitorCenter;
+
+// Create an Ice communicator to initialize the Ice runtime. The communicator is disposed before the program exits.
+using Ice.Communicator communicator = Ice.Util.initialize(ref args);
+
+// GreeterPrx is a class generated by the Slice compiler. We create a proxy from a communicator and a "stringified
+// proxy" with the address of the target object.
+// If you run the server on a different computer, replace localhost in the string below with the server's hostname
+// or IP address.
+GreeterPrx greeter = GreeterPrxHelper.createProxy(communicator, "greeter:tcp -h localhost -p 4061");
+
+// Create a proxy to the slow greeter. It uses the same connection is the regular greeter.
+GreeterPrx slowGreeter = GreeterPrxHelper.createProxy(communicator, "slowGreeter:tcp -h localhost -p 4061");
+
+// Send a request to the regular greeter and get the response.
+string greeting = await greeter.GreetAsync(Environment.UserName);
+Console.WriteLine(greeting);
+
+// Create another slow greeter proxy with an invocation timeout of 4 seconds (the default invocation timeout is
+// infinite).
+var slowGreeter4s = GreeterPrxHelper.uncheckedCast(slowGreeter.ice_invocationTimeout(4_000)); // in milliseconds
```

Comment:
We have a TimeSpan overload `ObjectPrx ice_invocationTimeout(TimeSpan newTimeout);` you should use it.